core/enumerator/each_with_index_spec.rb

```diff
@@ -19,18 +19,7 @@
     end.should raise_error(ArgumentError)
   end
 
-  it "passes on the given block's return value" do
-    arr = [1,2,3]
-    arr.delete_if.with_index { |a,b| false }
-    arr.should == [1,2,3]
-  end
-
-  it "returns the iterator's return value" do
-    [1,2,3].select.with_index { |a,b| false }.should == []
-  end
-
```

Comment:
Actually `#each_with_index` behaves in the same way, so it makes sense to keep these two cases here.

Reply:
Added this one back in, but changed to `each_with_index`. Added an extra test to check the true-block as well.

Reply:
Don't you think the second test `"passes on the given block's return value"` also makes sense to add back?

Reply:
That sound very sensible as well, added that one too.